### ISSUE TITLE
How can we select only year?

### ISSUE THREAD

**Issue by MohsinIkram-Auxilium** [Macacoazul01/month_picker_dialog#102]
Your package is great. 
But is there any way to select only Year from this calendar?
Like there is a some requirement to choose the only year. So I'm curious if there a way to select only Year?
Please guide. Thanks

**Comment by Macacoazul01** [comment]
Today it isnt, but i can update the package to do so

**Comment by Macacoazul01** [comment]
Technically you can select the year + month and use just the year part, but this isn't efficient when looking at the user's perspective

**Comment by MohsinIkram-Auxilium** [comment]
@Macacoazul01 Yes but there are some requirements in daily life to pick only month and only year.
So this package is great. Because it is included showMonthPicker & showMonthRangePicker.
It would be great if there is an option to select only year.
So people can user your package for the both Month and Year cases.
Instead of added two different packages for Month and Year.
Looking for your great support.

**Comment by Macacoazul01** [comment]
Avaliable in `5.0.0-dev.3` with the new `showYearPicker` function

**Comment by MohsinIkram-Auxilium** [comment]
Thanks for the quick response.
I have upgraded your package and set up all the things (except
localisation).
Because I'm using the Getx package for all the things including the
localisation. So I have not added the separate package for the
localisation.
But I'm facing this error. Before the new version it was working fine.
Please let me know how I can solve this with Getx.
Looking forward to your kind response.
Thanks

**Comment by Macacoazul01** [comment]
Can you send a full sample code?

**Comment by MohsinIkram-Auxilium** [comment]
@Gianluca Bettega Sorry. It is our company project.
Let me share the basic things

Here is the method I'm using to show the Month Picker

showMonthPickerDialog(BuildContext context, DateTime selectedDate,
    Function(DateTime)? onMonthSelected) {
  showMonthPicker(
    context: context,
    firstDate: DateTime.parse(UserPreferences.getJoiningDate()),
    lastDate: DateTime.now(),
    initialDate: selectedDate,
    monthPickerDialogSettings:  MonthPickerDialogSettings(
        headerSettings: const PickerHeaderSettings(
          headerBackgroundColor: AppColors.themeColor,
          headerCurrentPageTextStyle:
          TextStyle(fontSize: 14, color: Colors.white),
          headerSelectedIntervalTextStyle:
          TextStyle(fontSize: 16, color: Colors.white),
        ),
        dialogSettings: PickerDialogSettings(
            locale: const Locale('en'),
            dialogRoundedCornersRadius: 14,
            dialogBackgroundColor: Colors.blueGrey[50],
            yearFirst: false,
            dismissible: true

        ),
        buttonsSettings: PickerButtonsSettings(
          // buttonBorder: const CircleBorder(),
          buttonBorder: const RoundedRectangleBorder(),
          selectedMonthBackgroundColor: Colors.amber[900],
          selectedMonthTextColor: Colors.white,
          unselectedMonthsTextColor: Colors.black,
          currentMonthTextColor: AppColors.themeColor,
          yearTextStyle: const TextStyle(
            fontSize: 10,
          ),
          monthTextStyle: const TextStyle(
            fontWeight: FontWeight.bold,
            fontSize: 18,
          ),

        )),
    confirmWidget: const Text(
      'Ok',
      style: TextStyle(
        color: AppColors.themeColor,
        // fontFamily: 'Manrope',
        // fontWeight: FontWeight.w700,
      ),
    ),
    cancelWidget: const Text(
      'Cancel',
      style: TextStyle(
        // color: Colors.red[900],
        // fontFamily: 'Manrope',
        // fontWeight: FontWeight.w700,
      ),
    ),
    // headerTitle: const Text(
    //   'Month Picker Dialog',
    //   style: TextStyle(color: Colors.white),
    // ),
  )
      .then((DateTime? date) {
    if (date != null) {
      if (onMonthSelected != null) {
        onMonthSelected(date);
      }
    }
  });
}


Here is the Code in the Main class

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(
      translations: Languages(),
      fallbackLocale: const Locale('en', 'US'),
      debugShowCheckedModeBanner: false,
      initialRoute: "/splash",
      initialBinding: HomeBinding(),
      routes: {
        "/splash": (context) => const SplashScreen(),
      },
    );
  }
}

**Comment by Macacoazul01** [comment]
Can you run a flutter create and generate a project with this error?

**Comment by MohsinIkram-Auxilium** [comment]
Sorry @Gianluca, I'm stuck in the ongoing things and have not much time.
I will make the thing and share it with you in future. But for now I have
fixed the issue.
I just commented the line from this function and now it's working fine.
The line I commented -> "locale: const Locale('en')"

Thanks for your great support.

dialogSettings: PickerDialogSettings(
    // locale: const Locale('en'),
    dialogRoundedCornersRadius: 14,
    dialogBackgroundColor: Colors.blueGrey[50],
    yearFirst: false,
    dismissible: true),
